src/vstest.console/CommandLine/CommandLineOptions.cs

```diff
@@ -283,28 +284,32 @@ public void AddSource(string source)
             {
                 throw new CommandLineException(CommandLineResources.CannotBeNullOrEmpty);
             }
-
+            var filePatternParser = new FilePatternParser();
             source = source.Trim();
-
+            
             // Convert the relative path to absolute path
-            if(!Path.IsPathRooted(source))
+            if (!Path.IsPathRooted(source))
             {
                 source = Path.Combine(FileHelper.GetCurrentDirectory(), source);
             }
 
-            if (!FileHelper.Exists(source))
-            {
-                throw new CommandLineException(
-                    string.Format(CultureInfo.CurrentUICulture, CommandLineResources.TestSourceFileNotFound, source));
-            }
+            var sourceFiles = filePatternParser.GetMatchingFiles(source);
 
-            if (this.sources.Contains(source, StringComparer.OrdinalIgnoreCase))
+            foreach (var sourceFile in sourceFiles)
             {
-                throw new CommandLineException(
-                    string.Format(CultureInfo.CurrentCulture, CommandLineResources.DuplicateSource, source));
-            }
+                if (!FileHelper.Exists(sourceFile))
+                {
+                    throw new CommandLineException(
+                        string.Format(CultureInfo.CurrentUICulture, CommandLineResources.TestSourceFileNotFound, source));
+                }
```

Comment:
Do this only for valid source path and not for pattern

Reply:
Done